mmap-alloc/src/lib.rs

```diff
@@ -255,6 +231,9 @@ impl Default for MapAllocBuilder {
 pub struct MapAlloc {
     pagesize: usize,
     huge_pagesize: Option<usize>,
+    #[cfg_attr(target_os = "linux", allow(unused))] read: bool,
+    #[cfg_attr(target_os = "linux", allow(unused))] write: bool,
+    #[cfg_attr(target_os = "linux", allow(unused))] exec: bool,
```

Comment:
It seems unfortunate to have this duplication (the same info should be stored in the `perms` field. Could we either:

1. Use helper fns like `perms::can_read(Perm)`
2. Make Perm a real type (maybe using [bitflags](https://crates.io/crates/bitflags), with a method `to_os_perm()` which returns the right type depending on the OS? The custom type could support tests like `perm.contains(perms::READ)`.

Reply:
I agree that this is a tad ugly, but given that we almost exclusively use perms as opaque, and only use the boolean functionality in one place (realloc), I think it's better to keep it simple.